youtube_dl/extractor/clyp.py

```diff
@@ -0,0 +1,53 @@
+# coding: utf-8
+
+from __future__ import unicode_literals
+
+import re
+
+from .common import InfoExtractor
+
+
+class ClypIE(InfoExtractor):
+    _VALID_URL = r'https://clyp\.it/........'
+
+    _TESTS = [{
+        'url': 'https://clyp.it/ojz2wfah',
+        'md5': '1d4961036c41247ecfdcc439c0cddcbb',
+        'info_dict': {
+            'id': 'ojz2wfah',
+            'ext': 'mp3',
+            'title': 'Krisson80 - bits wip wip',
+            'description': '#Krisson80BitsWipWip #chiptune\n#wip',
+        },
+    }, {
+        'url': 'https://clyp.it/ojz2wfah',
+        'only_matching': True,
+    }]
+
+    def _real_extract(self, url):
+        video_id = url.replace("https://clyp.it/", "")
+        api_url = 'https://api.clyp.it/' + video_id
+        metadata = self._download_json(api_url, video_id)
+
+        title = metadata['Title']
+        description = metadata['Description']
```

Comment:
Every field apart from title should be treated as optional and extraction should not fail when it's missing.